core/src/test/scala/com/spotify/featran/transformers/NGramsTransformerSpec.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import org.scalacheck._
+
+object NGramsTransformerSpec extends TransformerProp("NHotEncoder") {
+
+  private implicit val labelArb = Arbitrary {
+    Gen.choose(1, 5).flatMap(Gen.listOfN(_, Gen.alphaStr))
+  }
+
+  property("default") = Prop.forAll { xs: List[List[String]] =>
+    val transformer = new NGramsTransformer("n_gram", " ", 2, 4)
+    val ngrams = xs.map(transformer.ngrams(_))
```

Comment:
I don't really like this test because it doesn't exercise the `ngrams` function independently, which seems harder to do in a property test. Ideally I'd have a separate unit test to check that in isolation. What do you think?

Reply:
It's OK for now. We can make assertions more black box as part of #53.